tests/ci_build/lint_cpp.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import os
+import re
+import sys
+
+import cpplint
+from cpplint import _cpplint_state
+
+CXX_SUFFIX = set(["cc", "c", "cpp", "h", "cu", "hpp"])
+
+
+def filepath_enumerate(paths):
+    """Enumerate the file paths of all subfiles of the list of paths"""
+    out = []
+    for path in paths:
+        if os.path.isfile(path):
+            out.append(path)
+        else:
+            for root, dirs, files in os.walk(path):
+                for name in files:
+                    out.append(os.path.normpath(os.path.join(root, name)))
+    return out
+
+
+def get_header_guard_dmlc(filename):
+    """Get Header Guard Convention for DMLC Projects.
+
+    For headers in include, directly use the path
+    For headers in src, use project name plus path
+
+    Examples: with project-name = dmlc
+        include/dmlc/timer.h -> DMLC_TIMTER_H_
+        src/io/libsvm_parser.h -> DMLC_IO_LIBSVM_PARSER_H_
+    """
+    fileinfo = cpplint.FileInfo(filename)
+    file_path_from_root = fileinfo.RepositoryName()
+    inc_list = ["include", "api", "wrapper", "contrib"]
+    if os.name == "nt":
+        inc_list.append("mshadow")
+
+    if file_path_from_root.find("src/") != -1 and _HELPER.project_name is not None:
+        idx = file_path_from_root.find("src/")
+        file_path_from_root = _HELPER.project_name + file_path_from_root[idx + 3 :]
+    else:
+        idx = file_path_from_root.find("include/")
+        if idx != -1:
+            file_path_from_root = file_path_from_root[idx + 8 :]
+        for spath in inc_list:
+            prefix = spath + "/"
+            if file_path_from_root.startswith(prefix):
+                file_path_from_root = re.sub("^" + prefix, "", file_path_from_root)
+                break
+    return re.sub(r"[-./\s]", "_", file_path_from_root).upper() + "_"
+
+
+class Lint:
+    def __init__(self):
+        self.project_name = "xgboost"
```

Comment:
```suggestion
class Lint:
    project_name = "xgboost"

    def __init__(self):
```

I suggest this change together with accessing the global `Lint.project_name` rather than the instance's `_HELPER.project_name`.